core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/solver/scope/SolverScope.java

```diff
@@ -210,6 +213,26 @@ public void setBestSolutionTimeMillis(Long bestSolutionTimeMillis) {
         this.bestSolutionTimeMillis = bestSolutionTimeMillis;
     }
 
+    /**
+     * See {@link #setWarmStart(boolean)}.
+     * Retrieve the value previously set, and set it to false afterwards.
+     *
+     * @return
+     */
+    public boolean isWarmStart() {
+        return isWarmStart.getAndSet(false);
+    }
+
+    /**
+     * Set to true after problem changes have just finished processing.
+     * Set to false once the solver has first started after that.
+     * 
+     * @param warmStart
+     */
+    public void setWarmStart(boolean warmStart) {
+        isWarmStart.set(warmStart);
+    }
+
```

Comment:
I'm not too excited about this. You could even call it a hack.
Got any better ideas?